### ISSUE TITLE
Docs incorrect for cf writer compression options

### ISSUE THREAD

**Issue by simonrp84** [pytroll/satpy#2386]
**Describe the bug**
The docs here: https://satpy.readthedocs.io/en/stable/api/satpy.writers.cf_writer.html
State that compression options can be set using, for example, `scn['my_dataset'].encoding = {'zlib': False}`
However, this appears to be incorrect. It should be `scn['my_dataset'].attrs['encoding'] = {'zlib': False}`

**Comment by djhoese** [comment]
As mentioned on slack, my best guess is this is a bug where any user provided compression automatically overwrites the encoding of the DataArray here:

https://github.com/pytroll/satpy/blob/0393daf22a20614915ff7ec4d8a8e0844ec8a515/satpy/writers/cf_writer.py#L663-L664

but the compression is on by default here:

https://github.com/pytroll/satpy/blob/0393daf22a20614915ff7ec4d8a8e0844ec8a515/satpy/writers/cf_writer.py#L575-L583

**Comment by mraspaud** [comment]
@sfinkens did you work on this a while ago?

**Comment by ghiggi** [comment]
To be sure the cause is not the `netcdf4` version, @simonrp84 which is the one you are using?

Since `netCDF4>=1.6.0`, the `zlib` key is deprecated in favor of the `compression` key. See: 
https://github.com/Unidata/netcdf4-python/blob/master/Changelog#L35
With the xarray [PR 6981](https://github.com/pydata/xarray/pull/6981/files) merged, now xarray expect/accept the `compression` key instead of `zlib`, and the `zlib` has no effect anymore.

If opened https://github.com/pytroll/satpy/issues/2244 some time ago to warn about this coming up.

**Comment by simonrp84** [comment]
I'm on netCDF4 v1.6.0:
```
(base) C:\Users\Simon>mamba list netcdf4
# packages in environment at C:\Users\Simon\miniconda3:
#
# Name                    Version                   Build  Channel
netcdf4                   1.6.0           nompi_py310h459bb5f_103    conda-forge
```

**Comment by ghiggi** [comment]
Et voila :D If you downgrade netCDF, the problem is temporary solved on your side ;)

**Comment by simonrp84** [comment]
Indeed! But as netCDF4 >=1.6.0 is the default now, the docs should reflect the current library rather than requiring an outdated one :-)

**Comment by djhoese** [comment]
> Indeed! But as netCDF4 >=1.6.0 is the default now, the docs should reflect the current library rather than requiring an outdated one :-)

And not only that, it should probably provide an automatic workaround and deprecation warning...or do we just depend on xarray to do that? There are a couple deprecations related to compression in the CF writer so maybe the fix for this issue (the doc updates) and the fix for #2244 (zlib having no effect) should include removing the deprecations and depend on xarray/NetCDF4 as much as possible to warn the user?

**Comment by sfinkens** [comment]
> @sfinkens did you work on this a while ago?

I can't remember :sweat_smile: But I have time to take a look this week. Long time no PR...
